test_conformance/api/test_api_min_max.cpp

```diff
@@ -491,47 +493,40 @@ int test_min_max_mem_alloc_size(cl_device_id deviceID, cl_context context, cl_co
     else
         requiredAllocSize = 128 * 1024 * 1024;
 
-    /* Get the max mem alloc size */
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_MAX_MEM_ALLOC_SIZE, sizeof( maxAllocSize ), &maxAllocSize, NULL );
-    test_error( error, "Unable to get max mem alloc size from device" );
+    /* Get the max mem alloc size, limit the alloc to half of the available
+     * memory */
+    maxAllocSize = get_device_info_max_mem_alloc_size(
+        deviceID, MAX_DEVICE_MEMORY_SIZE_DIVISOR);
+    memSize = get_device_info_global_mem_size(deviceID,
+                                              MAX_DEVICE_MEMORY_SIZE_DIVISOR);
 
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_GLOBAL_MEM_SIZE, sizeof( memSize ), &memSize, NULL );
-    test_error( error, "Unable to get global memory size from device" );
+    if (memSize < maxAllocSize)
+    {
+        log_info("Global memory size is less than max allocation size, using "
+                 "that.\n");
+        maxAllocSize = memSize;
+    }
 
     if (memSize > (cl_ulong)SIZE_MAX) {
       memSize = (cl_ulong)SIZE_MAX;
     }
 
-    if( maxAllocSize < requiredAllocSize)
+    if ((0 == gIsEmbedded && maxAllocSize < 128 * 1024 * 1024)
+        || maxAllocSize < 1 * 1024 * 1024)
```

Comment:
These checks can introduce new failures since maxAllocSize has already been reduced. There should be a clear distinction between reported max alloc size and any modifications we do to it. I still have the same objection about reducing allocation size in two different places. Applying this factor up front is reducing coverage for implementations that currently pass the test with their reported size--after this change they would be testing only half the size they're capable of.

Reply:
This value is 1M for embedded and 128M otherwise.
I changed the check to use requiredAllocSize for me consistency but I don't see an application reporting 2M or 256M which with the reduction would make it fail ?